04_numeronym/madhavansu/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"os"
+)
+
+var out io.Writer = os.Stdout
+
+func main() {
+	fmt.Fprint(out, numeronyms("accessibility", "Kubernetes", "abc"))
+}
+func numeronyms(vals ...string) []string {
+	var strcopy []string
```

Comment:
Since we know the length of the input array, it is a good practice is to pre-allocate the size of array `strcopy`.

Reply:
The functional implementation uses the append string, when you use append and also pre-allocate the size, it will start appending at the end of the allocated size and you will see an unnecessary space preponed to the string.

Reply:
Exactly, if we pre-allocate we don't need to append we can just assign the value at the position. As suggested in another comment.

Reply:
Or, you can preallocate `strcopy` with a size of 0 and capacity of len(vals) 
```go
strcopy := make([]string, 0, len(vals))
```
and continue appending values to it.

My thought here is when we don't init a slice with the required capacity and keep appending values to the slice everytime the size of slice exceeds its capacity it will have to 
a) reallocate a new slice with higher capacity b)copy its own values and c) perform append.
Maybe @juliaogris can confirm on this.

However for a smaller data set like this it shouldn't matter much :smiley:

Reply:
Updated